### DISCUSSION TITLE
Isn't it possible to contribute to `pocketbase/ui`?

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#1061] [Q&A]
![Screenshot 2022-11-18 074528](https://user-images.githubusercontent.com/44144724/202744668-dd3599d9-c22c-434e-b321-15a6430c3890.png)

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the question.

The Admin UI code could be found in https://github.com/pocketbase/pocketbase/tree/rc/ui.

If you are asking whether it is possible to extend the Admin UI with custom user code, then No, currently this is not possible (you can also check this issue - https://github.com/pocketbase/pocketbase/issues/898).

**Reply by babakfp** [reply]
Hey
What I mean is, how to pass the login page?

**Reply by ganigeorgiev** [reply]
I'm sorry, I still don't understand what actually you are asking.

When you start PocketBase for the first time and open the Admin UI, it will prompt you to create your first admin account.

If you are asking for how to access the live demo, aka. https://pocketbase.io/demo/, you can use:
```
Email: test@example.com
Pass:  123456
```

**Reply by babakfp** [reply]
Can I connect to the database of this website https://pocketbase.io/demo? Because running the Go code gives me an access denied error.

**Reply by ganigeorgiev** [reply]
> Can I connect to the database of this website https://pocketbase.io/demo?

Yes, you can connect to the live demo over its rest API, but please note that the address is `https://pocketbase.io` (the `/demo` is just a page with an iframe so that we can show the reset message):

```js
import PocketBase from "pocketbase";

const pb = new PocketBase("https://pocketbase.io");

const posts = await pb.collection("posts").getFullList();
```

> Because running the Go code gives me an access denied error.

What Go code?
Could you provide a code sample what you are trying to do?

**Reply by babakfp** [reply]
Ok so, let me explain it from the beginning. I cloned this repo `pocketbase/pocketbase` and now I just want to run `pocketbase/pocketbase/ui` and play with the code and do some changes to the frontend side. When I run `pnpm dev` and go to http://127.0.0.1:3000, it redirects me to http://127.0.0.1:3000/#/login. I want to pass this page and see the dashboard, the same as the demo... I need to be connected to a database right? I want to use the demo DB, but I tried doing this `PB_BACKEND_URL = "https://pocketbase.io"` and entering the demo login info in the login page but it didn't work (Invalid login credentials).

**Reply by ganigeorgiev** [reply]
Ah, I understand what do you mean now. You can find more documentation how to do frontend changes in the https://github.com/pocketbase/pocketbase/blob/master/CONTRIBUTING.md#making-changes-in-the-admin-ui.

In short, you need the API server running in the background on the default `localhost:8090` port OR create a `ui/.env.development.local` file with `PB_BACKEND_URL = https://pocketbase.io`.

**Reply by babakfp** [reply]
![Screenshot 2022-11-19 012402](https://user-images.githubusercontent.com/44144724/202844263-2453c5e7-e301-42bd-abae-547f3b043ac9.png)

![Screenshot 2022-11-19 012444](https://user-images.githubusercontent.com/44144724/202844265-301d4e75-8f39-4062-b93b-3905d38658ef.png)

![Screenshot 2022-11-19 012659](https://user-images.githubusercontent.com/44144724/202844316-1f559514-40dd-4a21-8ab8-abc090579663.png)

**Reply by ganigeorgiev** [reply]
@babakfp I forgot about the default CSP, sorry. I'll update it in the future, but for now if you want to use a development backend url different then localhost you can temporary disable it by going to `ui/index.html` and remove the following meta tag:

```html
    ...
    <meta
        http-equiv="Content-Security-Policy"
        content="default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' http://127.0.0.1:* data:; connect-src 'self' http://127.0.0.1:*; script-src 'self' 'sha256-GRUzBA7PzKYug7pqxv5rJaec5bwDCw1Vo6/IXwvD3Tc='"
    />
    ...
```

**Reply by babakfp** [reply]
After removing the meta tag, I can see the loading icon when I submit, but it still shows me the (Invalid login credentials) error.

**Reply by ganigeorgiev** [reply]
@babakfp I've just tested it locally and I cannot reproduce the error. Try refreshing the page and/or check in the browser's "Inspect tool > Network tab" the error request and response.

**Reply by babakfp** [reply]
From this login page: `http://127.0.0.1:3000/#/login`
Network tab:
![Screenshot 2022-11-19 020120](https://user-images.githubusercontent.com/44144724/202845462-6886f0f1-5435-4717-9810-2c51bb8cf3f1.png)

**Reply by ganigeorgiev** [reply]
https://pocketbase.io is running with PocketBase v0.8.0 ([it was released a couple hours ago](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0)).

I guess your repo is with v0.7.x. Please fetch the latest `master`, aka. `git pull origin master` and try again.

**Reply by babakfp** [reply]
👏👏 Success
Thanks for your help and quick replies.
